README.md

```diff
@@ -0,0 +1,80 @@
+# Nexmo JWT Generator for Ruby
+
+[![Gem Version](https://badge.fury.io/rb/nexmo-jwt.svg)](https://badge.fury.io/rb/nexmo-jwt) [![Build Status](https://api.travis-ci.org/Nexmo/nexmo-jwt-ruby.svg?branch=master)](https://travis-ci.org/Nexmo/nexmo-jwt-ruby) [![Coverage Status](https://coveralls.io/repos/github/Nexmo/nexmo-jwt-ruby/badge.svg?branch=coveralls)](https://coveralls.io/github/Nexmo/nexmo-jwt-ruby?branch=master)[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v2.0%20adopted-ff69b4.svg)](CODE_OF_CONDUCT.md)
+
+<img src="https://developer.nexmo.com/assets/images/Vonage_Nexmo.svg" height="48px" alt="Nexmo is now known as Vonage" />
+
+This is the Ruby library to generate Nexmo JSON Web Tokens (JWTs). To use it you'll
+need a Nexmo account. Sign up [for free at nexmo.com][signup].
+
+* [Requirements](#requirements)
+* [Installation](#installation)
+* [Usage](#usage)
+* [Documentation](#documentation)
+* [License](#license)
+
+## Requirements
+
+The JWT generator supports Ruby version 2.5 or newer.
+
+## Installation
+
+To install the Ruby client library using Rubygems:
+
+    gem install nexmo-jwt
+
+Alternatively you can clone the repository:
+
+    git clone git@github.com:Nexmo/nexmo-jwt-ruby.git
+
+## Usage
+
+By default the Nexmo JWT generator creates a short lived JWT (15 minutes) per request.
+To generate a long lived JWT for multiple requests, specify a longer value in the `exp`
+parameter during initialization. 
+
+Example with no custom configuration:
+
+```ruby
+@builder = Nexmo::JWTBuilder.new(application_id: YOUR_APPLICATION_ID, private_key: YOUR_PRIVATE_KEY)
+@token = @builder.jwt.generate
```

Comment:
What's `.jwt` here? I'd expect to call `@builder.generate_jwt`

@dragonmantank Can you reuse a builder later in the app, or should this be a one shot generation?

Reply:
It's very common in Ruby to separate out unique concerns into their own classes. In this case, there is the builder of the generator instance, and then there is the generation of a token with that instance. The latter only needs to know some of that information during usage, and it's easier to maintain the code when it's split out as such. That's why Ruby does it so frequently.

Reply:
We defined two patterns for use:

1. As an instantiated object, which can be re-used. This is why we expect a TTL instead of a hard expiration time. For example, You could use a single object to generate a token with 1 set of claims with a 30 second TTL, and then re-use the instantiated version to change the claims and generate a second token.
2. As a Factory, where every invocation requires a full set of options. The Factory would need all information (app_id, key, any additional claims) every time it is called.

Would the following code be legal?

```ruby
# Factory method to get a token in one shot
@token = Nexmo::JWTBuilder.new(application_id: YOUR_APPLICATION_ID, private_key: YOUR_PRIVATE_KEY).jwt.generate

# Instantiated object
@builder = Nexmo::JWTBuilder.new(application_id: YOUR_APPLICATION_ID, private_key: YOUR_PRIVATE_KEY)
@builder.ttl = 1500
@token = @builder.jwt.generate

@builder.ttl = 100
@token2 = @builder.jwt.generate
```

Reply:
Hi @dragonmantank, Yes the following code you enumerated works perfectly as described.